security/pkg/pki/ca/controller/secret.go

```diff
@@ -207,9 +210,17 @@ func (sc *SecretController) upsertSecret(saName, saNamespace string) {
 		PrivateKeyID: key,
 		RootCertID:   rootCert,
 	}
-	_, err = sc.core.Secrets(saNamespace).Create(secret)
+
+	// We retry several times when create secret.
+	for i := 0; i < secretCreationRetry; i++ {
+		_, err = sc.core.Secrets(saNamespace).Create(secret)
+		if err == nil {
+			break
```

Comment:
Otherwise print out some logs for debugging purpose?

Reply:
Good point, done.